src/models/train.py

```diff
@@ -90,7 +92,7 @@ def instantiate_trainer(self, **kwargs: Any) -> Trainer:
 
         if self.datamodule.val_path:
             if self.datamodule.is_classification:
-                checkpoint_metric = "val/roc_auc"
+                checkpoint_metric = "val/{}".format(self.config["fit"]["optimize"])
```

Comment:
Could we change this so that train metrics can also be passed? Sometimes we want to minimize for metrics like `train/loss`. 

You could just change it so that the config accepts something like `optimize: train/loss"